source/slang/ir-glsl-legalize.cpp

```diff
@@ -425,6 +502,16 @@ GLSLSystemValueInfo* getGLSLSystemValueInfo(
     }
     else if (semanticName == "nv_viewport_mask")
     {
+        // TODO: This doesn't seem to work correctly on it's own between hlsl/glsl
+
+        // Indeed on slang issue 109 claims this remains a problem  
+        // https://github.com/shader-slang/slang/issues/109
+
+        // On hlsl it's UINT related. "higher 16 bits for the right view, lower 16 bits for the left view."
+        // There is use in hlsl shader code as uint4 - not clear if that varies 
+        // https://github.com/KhronosGroup/GLSL/blob/master/extensions/nvx/GL_NVX_multiview_per_view_attributes.txt
+        // On glsl its highp int gl_ViewportMaskPerViewNV[];
```

Comment:
Yeah, we originally had a request to support the NVIDIA GLSL extensions for multi-view, but when I started to dig into them it began to appear that there wasn't really a close match between the HLSL NVAPI extension and the GLSL extension, so the conversion wouldn't really be easy.